sabnzbd/downloader.py

```diff
@@ -420,8 +425,30 @@ def run(self):
         # Kick BPS-Meter to check quota
         BPSMeter.do.update()
 
+        # Counts number of iterations with no articles found
+        idle_count = 0
+
+        # Store when each server did last search for articles
+        tested_time = {}
+
+        # Store when each server last downloaded anything or found an article
+        last_busy = {}
+
+        # Keeps track of how long one iteration takes, don't sleep if it takes too long
+        loop_time = time.time()
+
         while 1:
+            idle_count += 1
             for server in self.servers:
+                serverid = server.id
+                if server.busy_threads:
+                    last_busy[serverid] = time.time()
+
+                # Skip this server if idle for 4 seconds and has already been tested less than 1 second ago
+                if not server.article_queue and last_busy.get(serverid, 0) + 2 < time.time() and tested_time.get(serverid, 0) + 2 > time.time():
+                    continue
```

Comment:
Sorry, just noticed the comment is out of date. It will test a server for new articles once every other second if it hasn't been active (found any articles to download) for the last 2 seconds.